components/autocomplete/index.js

```diff
@@ -311,21 +318,20 @@ export class Autocomplete extends Component {
 		return { open, range, query };
 	}
 
-	search( event ) {
+	search() {
 		const { open: wasOpen, suppress: wasSuppress } = this.state;
 		const { completers } = this.props;
-		const container = event.target;
 		// ensure that the cursor location is unambiguous
-		const cursor = this.getCursor( container );
+		const cursor = this.getCursor( this.searchContainer );
 		if ( ! cursor ) {
 			return;
 		}
 		// look for the trigger prefix and search query just before the cursor location
-		const match = this.findMatch( container, cursor, completers, wasOpen );
+		const match = this.findMatch( this.searchContainer, cursor, completers, wasOpen );
 		const { open, query, range } = match || {};
 		// asynchronously load the options for the open completer
-		if ( open && ( ! wasOpen || open.idx !== wasOpen.idx ) ) {
-			this.loadOptions( open.idx );
+		if ( open && ( ! wasOpen || open.idx !== wasOpen.idx || '' !== query ) ) {
```

Comment:
I would leave it up to the completer as to if it returned any results for an empty query (for example with the forward-slash block completer I expect a list of commonly used blocks as soon as I type forward-slash). Instead I would change the check to `query !== wasQuery` and get the existing query value from the state (like `wasOpen`).

Reply:
yea, that makes more sense and is exactly what I am trying to achieve here - load options if the query has changed. I'll look at `wasOpen` and reproduce that logic.